src/styles.typ

```diff
@@ -148,6 +148,66 @@
   ),
 )
 
+#let _is-stroke-compatible-type(value) = {
+  return (type(value) in (stroke, color, length) or
```

Comment:
Should `gradient` and `tiling` be included as well?

Ref: https://docs.rs/typst-library/0.13.1/src/typst_library/visualize/stroke.rs.html#361-403

```rust

cast! {
    type Stroke,
    thickness: Length => Self { .. },
    color: Color => Self { .. },
    gradient: Gradient => Self { .. },
    tiling: Tiling => Self { .. },
    mut dict: Dict => {
        ..
        Self {
            paint,
            thickness,
            cap,
            join,
            dash,
            miter_limit: miter_limit.map(Scalar::new),
        }
    },
}
```